src/vehicle.cpp

```diff
@@ -1316,41 +1310,108 @@ void vehicle::use_controls()
     }
 }
 
-void vehicle::start_engine()
+bool vehicle::start_engine( const int e )
+{
+    if( !is_engine_on( e ) ) { return false; }
+
+    const vpart_info &einfo = part_info( engines[e] );
+
+    if( !fuel_left( einfo.fuel_type ) ) {
+        add_msg( _("The %s is out of %s."), einfo.name.c_str(), ammo_name( einfo.fuel_type ).c_str() );
+        return false;
+    }
+
+    const bool combustion = einfo.fuel_type == fuel_type_gasoline ||
+        einfo.fuel_type == fuel_type_diesel;
+    const double dmg = 1 - (parts[engines[e]].hp / einfo.durability);
+    const int engine_power = part_power( engines[e], true );
+
+    // Diesel engines are harder to start in cold weather
+    double cold_factor = 0;
+    if( einfo.fuel_type == fuel_type_diesel ) {
+        cold_factor = 1 - g->get_temperature() / 40;
+        if( cold_factor < 0 ) { cold_factor = 0; }
+        if( cold_factor > 1 ) { cold_factor = 1; }
+    }
+
+    // Start attempt takes time and makes noise
+    if( einfo.fuel_type != fuel_type_muscle ) {
+        int start_time = combustion ? 50 : 0;
+        start_time += engine_power < 1600 ? engine_power / 32 : 50;
+        start_time += 50 * dmg;
+        start_time += 20 * cold_factor;
+        g->u.moves -= start_time;
+
+        const point p = parts[engines[e]].mount;
+        int gx, gy;
+        coord_translate( p.x, p.y, gx, gy );
+        sounds::ambient_sound( global_x() + gx, global_y() + gy, start_time / 10, "" );
+    }
+
+    if( combustion ) {
+        // Small engines can be started without a battery (pull start or kick start)
+        if( engine_power >= 50 ) {
+            const int penalty = ((engine_power / 2) * dmg) + ((engine_power / 5) * cold_factor);
```

Comment:
With small integers you should do the multiply before the divide to avoid truncation: `(1 / 2) * 10` = 0 vs `(1 * 10) / 2 = 5`